common/lib/common-utils/src/idleTaskScheduler.ts

```diff
@@ -0,0 +1,67 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ITelemetryProperties } from "@fluidframework/common-definitions";
+
+
+interface Task {
+    handle: () => any,
+}
+
+
+/**
+ * Helper class that is used to schedule non-essential tasks
+ * Time measurements are in milliseconds as a floating point with a decimal
+ */
+export default class IdleTaskScheduler {
+    //List of tasks waiting to be run
+    private idleTaskList:  Task[] = [];
+    // A reference to the task currently being processed.
+    private taskHandle = 0;
+
+    /*
+    Add tasks to FIFO queue of tasks that are run during idle callback period.
+    Creates idle callback if it doesn't exist with a timeout of 1s.
+    */
+    public enqueueTask<T>(task: () => T){
+        let newTask: Task;
+        newTask = {
```

Comment:
I'm not sure why you need to wrap task here, i.e. you can directly push taks to queue as is (changing signatures and usages)